### PR TITLE
Wago automation controller support + additional 'concentration' sensor

### PR DESCRIPTION
**Pull request by amigne**
Support for Wago automation controller, with addition of a new 'concentration' sensor type.

Wago manufactures programmable programmable automation controllers that can be used to retrieve data from misc sensors, such as temperature, humidity, CO2 concentration, contacts, ...

This PR adds support for such Wago PFC controllers and includes a new 'concentration' sensor type for gaz/particles concentration (expressed in "ppm" unit).

The Wago PFC returns raw data in an array without any meta-information describing the type of measured value or description of the origin of the data. This has to be manually set in the config.php. This is not very clean, but this is the only solution that permits to add some meaning to the collected SNMP values.

The code in this PR is working, but it is still considered as in progress. The following has to be done before being approved and merged:
- [x] Add some meaning for 'state' type values;
- [x] Fix the Health / Overview page for displaying 'State' graphs;
- [x] Fix the Health / Overview page for displaying 'Concentration' graphs;
- [x] Add display of 'Concentration' sensors on the device Overview page;
- [x] Add test data for this new OS;
- [x] Create a documentation for the $config['os']['wago']['plc'][] settings.

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by murrant** [comment]
@amigne I know it might be more than you want to tackle.  But maybe LibreNMS needs a way to have sensors of "unknown" type which are later assigned a type by the user in the webui...  I know of a few more devices that have external sensors that do not report the type of sensor connected via snmp.

**Comment by amigne** [comment]
> @amigne I know it might be more than you want to tackle. But maybe LibreNMS needs a way to have sensors of "unknown" type which are later assigned a type by the user in the webui... I know of a few more devices that have external sensors that do not report the type of sensor connected via snmp.

@murrant This sounds to be an easier way to fix this issue (missing of sensor type from the device) from the user point of view than playing with the configuration file. But I see a point anyway: having only the webgui as a way to set sensor types would be quite a nightmare if the user has about 50 devices with 10-15 sensors each (this is my case!). It would then need to manually edit the settings for 500-750 sensors through the webgui, which is not very convenient. If devices and sensors deployment is done with some reasonable thinking, it may be much easier to define a generic setting using the configuration file and to handle exceptions either with configuration file (using regexp), or with the webgui.

As I'm currently writing this module for my business, I propose to continue my current development based on the configuration file only... Once it is finished (and after my summer holidays are over ;-) ), I can start working on the webgui to manually define the sensor type for these devices.

**Comment by murrant** [comment]
Sure, but I don't think the config setting is going to be useful for most people.  What happens when you have a single device in a different configuration?

An option would be to have a button that says apply these sensor types to other devices (when clicked it shows a selectable list with all devices of the same OS)

**Comment by amigne** [comment]
> An option would be to have a button that says apply these sensor types to other devices (when clicked it shows a selectable list with all devices of the same OS)

Good idea : This is noted down for the upcoming webgui dev.

**Comment by amigne** [comment]
An issue still needs to be fixed : the discovery script creates new records in the 'sensors' DB table.

**Comment by amigne** [comment]
Issue fixed and successfully tested.

File: doc/Support/Configuration.md [hunk 1]
```diff
@@ -715,6 +715,93 @@ $config['ipmi']['type'][] = "open";
 
 [Distributed Poller](../Extensions/Distributed-Poller.md)
 
+### Wago data collection
```
Comment (murrant) [inline doc/Support/Configuration.md]: This should be under doc/Support/Device-Notes/

Comment (murrant) [reply doc/Support/Configuration.md]: Also, I still wouldn't mind a general approach.

Comment (amigne) [reply doc/Support/Configuration.md]: Well Tony, I'll move the doc at the right place. Thank you for the feedback.

Regarding the general approach, this is in my TODO list, but this needs more thinking to design a really generic form that could be used for this Wago device, as well as any new device. I'll start working on it as soon as I'll be back from holidays. 🌞

**Comment by Jellyfrog** [comment]
Closing due to inactivity. Feel free to reopen!

**Pull request closed without merging**